Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -477,6 +477,56 @@ void TextureCacheBase::BindTextures()
   }
 }
 
+struct ManualMipmapDetector
+{
+
+  ManualMipmapDetector()
+  {
+    min = { 255, 255, 255, 255 };
+    max = { 0, 0, 0, 0 };
+  }
+
+  void AddLevel(u32 width, u32 height, u32 row_length, const u8* buffer)
+  {
+    ++levels;
+
+    u64 avg[4] = { 0 };
+    for (u32 i = 0; i < height; ++i)
+    {
+      auto* row = buffer;
+      for (u32 j = 0; j < width; ++j)
+      {
+        avg[0] += row[0];
```

Comment:
This can just be:

```cpp
std::transform(avg.begin(), avg.end(), row, avg.begin(), std::plus<>());
```

can it not?

Reply:
Honestly I find that less readable, the for loop is pretty explicit about what it does and how it's compiled. I wouldn't recommend using that much template magic to replace this... who knows what it compiles to on all platforms.